geopandas/tests/test_sindex.py

```diff
@@ -18,20 +18,34 @@
 import numpy as np
 
 
-@pytest.mark.skipif(sindex.has_sindex(), reason="Spatial index present, skipping")
 class TestNoSindex:
-    def test_no_sindex(self):
-        """Checks that a warning is given when no spatial index is present."""
-        with pytest.warns(UserWarning):
+    @pytest.mark.skipif(sindex.has_sindex(), reason="Spatial index present, skipping")
+    def test_no_sindex_installed(self):
+        """Checks that an error is raised when no spatial index is present."""
+        with pytest.raises(RuntimeError):
             sindex.get_sindex_class()
 
+    @pytest.mark.skipif(
+        compat.HAS_RTREE or not compat.HAS_PYGEOS,
+        reason="rtree cannot be disabled via flags",
+    )
+    def test_no_sindex_active(self):
+        """Checks that an error is given when rtree is not installed
+        and compat.USE_PYGEOS is False.
+        """
+        state = compat.USE_PYGEOS  # try to save state
+        compat.set_use_pygeos(False)
+        with pytest.raises(RuntimeError):
+            sindex.get_sindex_class()
+        compat.set_use_pygeos(state)  # try to restore state
```

Comment:
I'm not sure that this will ever run, I don't think we have a CI run that has PyGEOS but not rtree? Can we have some type of parametrized CI run that runs with all combinations of optional dependencies (it would take a _long_ time to test)?